index.js

```diff
@@ -12,10 +12,27 @@ const makeDir = require('make-dir');
 const pify = require('pify');
 const pEvent = require('p-event');
 const fileType = require('file-type');
+const extName = require('ext-name');
 
 const fsP = pify(fs);
 const filenameFromPath = res => path.basename(url.parse(res.requestUrl).pathname);
 
+const getExtFromMime = res => {
+	const header = res.headers['content-type'];
+
+	if (!header) {
+		return null;
+	}
+
+	const exts = extName.mime(header);
+
+	if (exts.length !== 1) {
```

Comment:
Looking through `mime-db`, it seems it's returning the "right" extension first in most cases, but there are also cases like this:

````json
  "application/x-msdownload": {
    "source": "apache",
    "extensions": ["exe","dll","com","bat","msi"]
},
````

So I guess it's fine for those cases to fail.